### DISCUSSION TITLE
How to install the same package with two different architectures

### DISCUSSION THREAD

**Discussion by MicheleD1999** [Homebrew/discussions#5073] [Getting started] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.4
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1c612baa096c69f2fcb221c74e6f5b9979efdcee
Last commit: 7 days ago
Core tap JSON: 22 Jan 10:45 UTC
Core cask tap JSON: 22 Jan 10:45 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.1.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit westmere
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.2.1-x86_64
CLT: 15.1.0.0.1.1700200546
Xcode: 15.2
Rosetta 2: true
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libcln.6.dylib
  /usr/local/lib/libfirefly.dylib
  /usr/local/lib/libflint.18.0.dylib
  /usr/local/lib/libginac.11.dylib
  /usr/local/lib/libgmp.10.dylib
  /usr/local/lib/libhistory.8.2.dylib
  /usr/local/lib/libmpfr.6.dylib
  /usr/local/lib/libreadline.8.2.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/cln/GV.h
  /usr/local/include/cln/GV_complex.h
  /usr/local/include/cln/GV_integer.h
  /usr/local/include/cln/GV_modinteger.h
  /usr/local/include/cln/GV_number.h
  /usr/local/include/cln/GV_rational.h
  /usr/local/include/cln/GV_real.h
  /usr/local/include/cln/SV.h
  /usr/local/include/cln/SV_complex.h
  /usr/local/include/cln/SV_integer.h
  /usr/local/include/cln/SV_number.h
  /usr/local/include/cln/SV_rational.h
  /usr/local/include/cln/SV_real.h
  /usr/local/include/cln/SV_ringelt.h
  /usr/local/include/cln/V.h
  /usr/local/include/cln/cln.h
  /usr/local/include/cln/complex.h
  /usr/local/include/cln/complex_class.h
  /usr/local/include/cln/complex_io.h
  /usr/local/include/cln/complex_ring.h
  /usr/local/include/cln/condition.h
  /usr/local/include/cln/config.h
  /usr/local/include/cln/dfloat.h
  /usr/local/include/cln/dfloat_class.h
  /usr/local/include/cln/dfloat_io.h
  /usr/local/include/cln/exception.h
  /usr/local/include/cln/ffloat.h
  /usr/local/include/cln/ffloat_class.h
  /usr/local/include/cln/ffloat_io.h
  /usr/local/include/cln/float.h
  /usr/local/include/cln/float_class.h
  /usr/local/include/cln/float_io.h
  /usr/local/include/cln/floatformat.h
  /usr/local/include/cln/host_cpu.h
  /usr/local/include/cln/input.h
  /usr/local/include/cln/integer.h
  /usr/local/include/cln/integer_class.h
  /usr/local/include/cln/integer_io.h
  /usr/local/include/cln/integer_ring.h
  /usr/local/include/cln/intparam.h
  /usr/local/include/cln/io.h
  /usr/local/include/cln/lfloat.h
  /usr/local/include/cln/lfloat_class.h
  /usr/local/include/cln/lfloat_io.h
  /usr/local/include/cln/malloc.h
  /usr/local/include/cln/modinteger.h
  /usr/local/include/cln/modules.h
  /usr/local/include/cln/null_ring.h
  /usr/local/include/cln/number.h
  /usr/local/include/cln/number_io.h
  /usr/local/include/cln/numtheory.h
  /usr/local/include/cln/object.h
  /usr/local/include/cln/output.h
  /usr/local/include/cln/proplist.h
  /usr/local/include/cln/random.h
  /usr/local/include/cln/rational.h
  /usr/local/include/cln/rational_class.h
  /usr/local/include/cln/rational_io.h
  /usr/local/include/cln/rational_ring.h
  /usr/local/include/cln/real.h
  /usr/local/include/cln/real_class.h
  /usr/local/include/cln/real_io.h
  /usr/local/include/cln/real_ring.h
  /usr/local/include/cln/ring.h
  /usr/local/include/cln/sfloat.h
  /usr/local/include/cln/sfloat_class.h
  /usr/local/include/cln/sfloat_io.h
  /usr/local/include/cln/string.h
  /usr/local/include/cln/symbol.h
  /usr/local/include/cln/timing.h
  /usr/local/include/cln/types.h
  /usr/local/include/cln/univpoly.h
  /usr/local/include/cln/univpoly_complex.h
  /usr/local/include/cln/univpoly_integer.h
  /usr/local/include/cln/univpoly_modint.h
  /usr/local/include/cln/univpoly_rational.h
  /usr/local/include/cln/univpoly_real.h
  /usr/local/include/cln/version.h
  /usr/local/include/firefly/tinydir.h
  /usr/local/include/flint/NTL-interface.h
  /usr/local/include/flint/acb.h
  /usr/local/include/flint/acb_calc.h
  /usr/local/include/flint/acb_dft.h
  /usr/local/include/flint/acb_dirichlet.h
  /usr/local/include/flint/acb_elliptic.h
  /usr/local/include/flint/acb_hypgeom.h
  /usr/local/include/flint/acb_mat.h
  /usr/local/include/flint/acb_modular.h
  /usr/local/include/flint/acb_poly.h
  /usr/local/include/flint/acb_theta.h
  /usr/local/include/flint/acb_types.h
  /usr/local/include/flint/acf.h
  /usr/local/include/flint/acf_types.h
  /usr/local/include/flint/aprcl.h
  /usr/local/include/flint/arb.h
  /usr/local/include/flint/arb_calc.h
  /usr/local/include/flint/arb_fmpz_poly.h
  /usr/local/include/flint/arb_fpwrap.h
  /usr/local/include/flint/arb_hypgeom.h
  /usr/local/include/flint/arb_mat.h
  /usr/local/include/flint/arb_poly.h
  /usr/local/include/flint/arb_types.h
  /usr/local/include/flint/arf.h
  /usr/local/include/flint/arf_types.h
  /usr/local/include/flint/arith.h
  /usr/local/include/flint/bernoulli.h
  /usr/local/include/flint/bool_mat.h
  /usr/local/include/flint/ca.h
  /usr/local/include/flint/ca_ext.h
  /usr/local/include/flint/ca_field.h
  /usr/local/include/flint/ca_mat.h
  /usr/local/include/flint/ca_poly.h
  /usr/local/include/flint/ca_vec.h
  /usr/local/include/flint/calcium.h
  /usr/local/include/flint/crt_helpers.h
  /usr/local/include/flint/d_mat.h
  /usr/local/include/flint/d_vec.h
  /usr/local/include/flint/dirichlet.h
  /usr/local/include/flint/dlog.h
  /usr/local/include/flint/double_extras.h
  /usr/local/include/flint/double_interval.h
  /usr/local/include/flint/exception.h
  /usr/local/include/flint/fexpr.h
  /usr/local/include/flint/fexpr_builtin.h
  /usr/local/include/flint/fft.h
  /usr/local/include/flint/fft_small.h
  /usr/local/include/flint/fft_tuning.h
  /usr/local/include/flint/flint-config.h
  /usr/local/include/flint/flint.h
  /usr/local/include/flint/fmpq.h
  /usr/local/include/flint/fmpq_mat.h
  /usr/local/include/flint/fmpq_mpoly.h
  /usr/local/include/flint/fmpq_mpoly_factor.h
  /usr/local/include/flint/fmpq_poly.h
  /usr/local/include/flint/fmpq_types.h
  /usr/local/include/flint/fmpq_vec.h
  /usr/local/include/flint/fmpz.h
  /usr/local/include/flint/fmpz_extras.h
  /usr/local/include/flint/fmpz_factor.h
  /usr/local/include/flint/fmpz_lll.h
  /usr/local/include/flint/fmpz_mat.h
  /usr/local/include/flint/fmpz_mod.h
  /usr/local/include/flint/fmpz_mod_mat.h
  /usr/local/include/flint/fmpz_mod_mpoly.h
  /usr/local/include/flint/fmpz_mod_mpoly_factor.h
  /usr/local/include/flint/fmpz_mod_poly.h
  /usr/local/include/flint/fmpz_mod_poly_factor.h
  /usr/local/include/flint/fmpz_mod_types.h
  /usr/local/include/flint/fmpz_mod_vec.h
  /usr/local/include/flint/fmpz_mpoly.h
  /usr/local/include/flint/fmpz_mpoly_factor.h
  /usr/local/include/flint/fmpz_mpoly_q.h
  /usr/local/include/flint/fmpz_poly.h
  /usr/local/include/flint/fmpz_poly_factor.h
  /usr/local/include/flint/fmpz_poly_mat.h
  /usr/local/include/flint/fmpz_poly_q.h
  /usr/local/include/flint/fmpz_types.h
  /usr/local/include/flint/fmpz_vec.h
  /usr/local/include/flint/fmpzi.h
  /usr/local/include/flint/fq.h
  /usr/local/include/flint/fq_default.h
  /usr/local/include/flint/fq_default_mat.h
  /usr/local/include/flint/fq_default_poly.h
  /usr/local/include/flint/fq_default_poly_factor.h
  /usr/local/include/flint/fq_embed.h
  /usr/local/include/flint/fq_embed_templates.h
  /usr/local/include/flint/fq_mat.h
  /usr/local/include/flint/fq_mat_templates.h
  /usr/local/include/flint/fq_nmod.h
  /usr/local/include/flint/fq_nmod_embed.h
  /usr/local/include/flint/fq_nmod_mat.h
  /usr/local/include/flint/fq_nmod_mpoly.h
  /usr/local/include/flint/fq_nmod_mpoly_factor.h
  /usr/local/include/flint/fq_nmod_poly.h
  /usr/local/include/flint/fq_nmod_poly_factor.h
  /usr/local/include/flint/fq_nmod_types.h
  /usr/local/include/flint/fq_nmod_vec.h
  /usr/local/include/flint/fq_poly.h
  /usr/local/include/flint/fq_poly_factor.h
  /usr/local/include/flint/fq_poly_factor_templates.h
  /usr/local/include/flint/fq_poly_templates.h
  /usr/local/include/flint/fq_templates.h
  /usr/local/include/flint/fq_types.h
  /usr/local/include/flint/fq_vec.h
  /usr/local/include/flint/fq_vec_templates.h
  /usr/local/include/flint/fq_zech.h
  /usr/local/include/flint/fq_zech_embed.h
  /usr/local/include/flint/fq_zech_mat.h
  /usr/local/include/flint/fq_zech_mpoly.h
  /usr/local/include/flint/fq_zech_mpoly_factor.h
  /usr/local/include/flint/fq_zech_poly.h
  /usr/local/include/flint/fq_zech_poly_factor.h
  /usr/local/include/flint/fq_zech_types.h
  /usr/local/include/flint/fq_zech_vec.h
  /usr/local/include/flint/gmpcompat.h
  /usr/local/include/flint/gr.h
  /usr/local/include/flint/gr_generic.h
  /usr/local/include/flint/gr_mat.h
  /usr/local/include/flint/gr_mpoly.h
  /usr/local/include/flint/gr_poly.h
  /usr/local/include/flint/gr_special.h
  /usr/local/include/flint/gr_vec.h
  /usr/local/include/flint/hypgeom.h
  /usr/local/include/flint/limb_types.h
  /usr/local/include/flint/long_extras.h
  /usr/local/include/flint/longlong.h
  /usr/local/include/flint/longlong_asm_clang.h
  /usr/local/include/flint/longlong_asm_gcc.h
  /usr/local/include/flint/longlong_div_gnu.h
  /usr/local/include/flint/longlong_msc_arm64.h
  /usr/local/include/flint/longlong_msc_x86.h
  /usr/local/include/flint/machine_vectors.h
  /usr/local/include/flint/mag.h
  /usr/local/include/flint/mpf-impl.h
  /usr/local/include/flint/mpfr_mat.h
  /usr/local/include/flint/mpfr_vec.h
  /usr/local/include/flint/mpn_extras.h
  /usr/local/include/flint/mpoly.h
  /usr/local/include/flint/mpoly_types.h
  /usr/local/include/flint/n_poly.h
  /usr/local/include/flint/nf.h
  /usr/local/include/flint/nf_elem.h
  /usr/local/include/flint/nmod.h
  /usr/local/include/flint/nmod_mat.h
  /usr/local/include/flint/nmod_mpoly.h
  /usr/local/include/flint/nmod_mpoly_factor.h
  /usr/local/include/flint/nmod_poly.h
  /usr/local/include/flint/nmod_poly_factor.h
  /usr/local/include/flint/nmod_poly_mat.h
  /usr/local/include/flint/nmod_types.h
  /usr/local/include/flint/nmod_vec.h
  /usr/local/include/flint/padic.h
  /usr/local/include/flint/padic_mat.h
  /usr/local/include/flint/padic_poly.h
  /usr/local/include/flint/partitions.h
  /usr/local/include/flint/perm.h
  /usr/local/include/flint/profiler.h
  /usr/local/include/flint/qadic.h
  /usr/local/include/flint/qfb.h
  /usr/local/include/flint/qqbar.h
  /usr/local/include/flint/qsieve.h
  /usr/local/include/flint/templates.h
  /usr/local/include/flint/thread_pool.h
  /usr/local/include/flint/thread_support.h
  /usr/local/include/flint/ulong_extras.h
  /usr/local/include/ginac/add.h
  /usr/local/include/ginac/archive.h
  /usr/local/include/ginac/assertion.h
  /usr/local/include/ginac/basic.h
  /usr/local/include/ginac/class_info.h
  /usr/local/include/ginac/clifford.h
  /usr/local/include/ginac/color.h
  /usr/local/include/ginac/compiler.h
  /usr/local/include/ginac/constant.h
  /usr/local/include/ginac/container.h
  /usr/local/include/ginac/ex.h
  /usr/local/include/ginac/excompiler.h
  /usr/local/include/ginac/expair.h
  /usr/local/include/ginac/expairseq.h
  /usr/local/include/ginac/exprseq.h
  /usr/local/include/ginac/factor.h
  /usr/local/include/ginac/fail.h
  /usr/local/include/ginac/fderivative.h
  /usr/local/include/ginac/flags.h
  /usr/local/include/ginac/function.h
  /usr/local/include/ginac/ginac.h
  /usr/local/include/ginac/hash_map.h
  /usr/local/include/ginac/idx.h
  /usr/local/include/ginac/indexed.h
  /usr/local/include/ginac/inifcns.h
  /usr/local/include/ginac/integral.h
  /usr/local/include/ginac/integration_kernel.h
  /usr/local/include/ginac/lst.h
  /usr/local/include/ginac/matrix.h
  /usr/local/include/ginac/mul.h
  /usr/local/include/ginac/ncmul.h
  /usr/local/include/ginac/normal.h
  /usr/local/include/ginac/numeric.h
  /usr/local/include/ginac/operators.h
  /usr/local/include/ginac/parse_context.h
  /usr/local/include/ginac/parser.h
  /usr/local/include/ginac/power.h
  /usr/local/include/ginac/print.h
  /usr/local/include/ginac/pseries.h
  /usr/local/include/ginac/ptr.h
  /usr/local/include/ginac/registrar.h
  /usr/local/include/ginac/relational.h
  /usr/local/include/ginac/structure.h
  /usr/local/include/ginac/symbol.h
  /usr/local/include/ginac/symmetry.h
  /usr/local/include/ginac/tensor.h
  /usr/local/include/ginac/version.h
  /usr/local/include/ginac/wildcard.h
  /usr/local/include/gmp.h
  /usr/local/include/mpf2mpfr.h
  /usr/local/include/mpfr.h
  /usr/local/include/readline/chardefs.h
  /usr/local/include/readline/history.h
  /usr/local/include/readline/keymaps.h
  /usr/local/include/readline/readline.h
  /usr/local/include/readline/rlconf.h
  /usr/local/include/readline/rlstdc.h
  /usr/local/include/readline/rltypedefs.h
  /usr/local/include/readline/tilde.h

Warning: Unbrewed '.la' files were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected '.la' files:
  /usr/local/lib/libcln.la
  /usr/local/lib/libginac.la
  /usr/local/lib/libgmp.la
  /usr/local/lib/libmpfr.la

Warning: Unbrewed '.pc' files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected '.pc' files:
  /usr/local/lib/pkgconfig/cln.pc
  /usr/local/lib/pkgconfig/firefly.pc
  /usr/local/lib/pkgconfig/flint.pc
  /usr/local/lib/pkgconfig/ginac.pc
  /usr/local/lib/pkgconfig/gmp.pc
  /usr/local/lib/pkgconfig/history.pc
  /usr/local/lib/pkgconfig/mpfr.pc
  /usr/local/lib/pkgconfig/readline.pc

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libcln.a
  /usr/local/lib/libfirefly.a
  /usr/local/lib/libginac.a
  /usr/local/lib/libgmp.a
  /usr/local/lib/libhistory.a
  /usr/local/lib/libmpfr.a
  /usr/local/lib/libreadline.a

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  gmp
  cmake
  mpfr
  python@3.12
  cln
  ninja
  meson
  flint
  ginac

Warning: Your Homebrew's prefix is not /opt/homebrew.

Many of Homebrew's bottles (binary packages) can only be used with the default prefix.
Consider uninstalling Homebrew and reinstalling into the default prefix.
It is expected behaviour that some formulae will fail to build in this unsupported configuration.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

### Description of issue

Hello to everybody. I'm currently facing a problem of architectures on my macOS Sonoma 14.2.1. I need to install a package (named Kira, https://gitlab.com/kira-pyred/kira/-/tree/master) that for different reasons forces me to use an arch x86_64 for the installation (I'm trying to follow this procedure https://gitlab.com/kira-pyred/kira/-/issues/46#note_1452046671). 

The problem is that this package has several dependencies, like gcc, ginac, cln, meson, ninja and gmp. During my previous attempts, I installed these packages with both "brew install" command or from their git repositories and following the instructions. Right now, several of the dependencies are under the path /opt/homebrew.

As a consequence, I tried to install homebrew with x86_64, following the instructions in https://stackoverflow.com/questions/64951024/how-can-i-run-two-isolated-installations-of-homebrew, and I have now added to my .zshrc:
alias brew86='arch --x86_64 /usr/local/Homebrew/bin/brew'

My problem is that  if I try to install Kira package with the tap provided in
https://github.com/tueda/homebrew-loops
I obtain as error:

Last 15 lines from /Users/micheledradi/Library/Logs/Homebrew/kira/01.meson:
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/local/opt/meson/bin/meson", line 33, in <module>
    sys.exit(load_entry_point('meson==1.3.1', 'console_scripts', 'meson')())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/opt/meson/bin/meson", line 22, in importlib_load_entry_point
    for entry_point in distribution(dist_name).entry_points
                       ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/python@3.12/3.12.1_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/importlib/metadata/__init__.py", line 861, in distribution
    return Distribution.from_name(distribution_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/python@3.12/3.12.1_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/importlib/metadata/__init__.py", line 399, in from_name
    raise PackageNotFoundError(name)
importlib.metadata.PackageNotFoundError: No package metadata was found for meson
If reporting this issue please do so at (not Homebrew/brew or Homebrew/homebrew-core):
  https://github.com/tueda/homebrew-loops/issues
These open issues may also help:
ci: brew test kira fails on Linux https://github.com/tueda/homebrew-loops/issues/20

If I understand good, the problem is due to my previous dependencies installations under /opt/homebrew without x86_64 architecture, but the errors do not disappear even if I try to do brew uninstall <package>. In addition, if I try to reinstall them with brew86 install <package>, they return to /opt/homebrew and not /usr/local/Homebrew.

Is it a problem of PATH? Can I install two versions of the same package, one with brew, another with brew86? And how can I switch to use one or another?

Thanks to everybody,
Michele

**Comment by SMillerDev** [comment]
> If I understand good, the problem is due to my previous dependencies installations under /opt/homebrew without x86_64 architecture, but the errors do not disappear even if I try to do brew uninstall . In addition, if I try to reinstall them with brew86 install , they return to /opt/homebrew and not /usr/local/Homebrew.

It doesn't refer to any path under `/opt/homebrew` in the error, it's much more likely that the install just didn't work for unrelated reasons.

**Reply by MicheleD1999** [reply]
At this point, the only thing that I can do is install the package without the tap. More precisely, when I do arch --x86_64 ./configure I obtain that gmp package is not found
checking for __gmpz_init in -lgmp... no
configure: error: GNU MP not found, see http://gmplib.org/

But if I do brew list, gmp is present, and if I do 
brew86 install gmp
I obtain:
Running `brew update --auto-update`...
Warning: gmp 6.3.0 is already installed and up-to-date.
To reinstall 6.3.0, run:
  brew reinstall gmp

Is it possible that the problem is that gmp has not been compiled with arch -x86_64 and consequently not recognized? Since I need it also with the standard architecture, is it possible to install another gmp package inside /usr/local and point to it for my Kira installation?

Thank you,
Michele

**Reply by SMillerDev** [reply, marked as answer]
I just made a pull request to the tap to make some improvements, you could try again there

**Reply by MicheleD1999** [reply]
Thank you very much!

**Reply by MicheleD1999** [reply]
I tried with the new version of the tap and the error is apparently linked with the correct way of linking meson:

Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink share/vim/vimfiles/ftdetect
/usr/local/share/vim/vimfiles is not writable.


In addition to the previous error reported in the issue.

Thank you again,
Michele

**Reply by SMillerDev** [reply]
What does this say?

`ls -al /usr/local/share/vim/vimfiles`

And can you post the full error output?

**Reply by MicheleD1999** [reply]
Sure!

ls -al /usr/local/share/vim/vimfiles
total 0
drwxr-xr-x  4 root  wheel  128 19 Gen 18:50 .
drwxr-xr-x  3 root  wheel   96 19 Gen 18:50 ..
drwxr-xr-x  3 root  wheel   96 19 Gen 18:50 indent
drwxr-xr-x  3 root  wheel   96 19 Gen 18:50 syntax


While the total output trying to use the tap is:

brew86 install kira.rb
==> Downloading https://formulae.brew.sh/api/formula.jws.json
############################################################################################################################################ 100.0%
==> Downloading https://formulae.brew.sh/api/cask.jws.json
############################################################################################################################################ 100.0%
==> Fetching dependencies for tueda/loops/kira: meson
==> Fetching meson
==> Downloading https://ghcr.io/v2/homebrew/core/meson/manifests/1.3.1
Already downloaded: /Users/micheledradi/Library/Caches/Homebrew/downloads/4c9be34278d9d19e2839fce53a752d5084b2deb16aeb1502a21ed739c015795e--meson-1.3.1.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/meson/blobs/sha256:949ec0bb552146f962e41086b6b827e6d1bc7d130616361c472d1313afdd257f
Already downloaded: /Users/micheledradi/Library/Caches/Homebrew/downloads/7e5310b24c347708caeed73c5bbe4d6e0ba43fe2676636af60a44238bc588178--meson--1.3.1.all.bottle.tar.gz
==> Fetching tueda/loops/kira
==> Downloading https://gitlab.com/kira-pyred/kira/-/archive/2.3/kira-2.3.tar.gz
Already downloaded: /Users/micheledradi/Library/Caches/Homebrew/downloads/912c4eee6ca2c949bdd867034cec44a3056762091f893b660f3bd475b558b2e4--kira-2.3.tar.gz
==> Installing kira from tueda/loops
==> Installing dependencies for tueda/loops/kira: meson
==> Installing tueda/loops/kira dependency: meson
==> Downloading https://ghcr.io/v2/homebrew/core/meson/manifests/1.3.1
Already downloaded: /Users/micheledradi/Library/Caches/Homebrew/downloads/4c9be34278d9d19e2839fce53a752d5084b2deb16aeb1502a21ed739c015795e--meson-1.3.1.bottle_manifest.json
==> Pouring meson--1.3.1.all.bottle.tar.gz
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink share/vim/vimfiles/ftdetect
/usr/local/share/vim/vimfiles is not writable.

You can try again using:
  brew link meson
==> Summary
🍺  /usr/local/Cellar/meson/1.3.1: 250 files, 3.7MB
==> Installing tueda/loops/kira
==> meson --force-fallback-for=firefly ..
Last 15 lines from /Users/micheledradi/Library/Logs/Homebrew/kira/01.meson:
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/opt/meson/bin/meson", line 33, in <module>
    sys.exit(load_entry_point('meson==1.3.1', 'console_scripts', 'meson')())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/opt/meson/bin/meson", line 22, in importlib_load_entry_point
    for entry_point in distribution(dist_name).entry_points
                       ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/python@3.12/3.12.1_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/importlib/metadata/__init__.py", line 861, in distribution
    return Distribution.from_name(distribution_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/python@3.12/3.12.1_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/importlib/metadata/__init__.py", line 399, in from_name
    raise PackageNotFoundError(name)
importlib.metadata.PackageNotFoundError: No package metadata was found for meson

If reporting this issue please do so at (not Homebrew/brew or Homebrew/homebrew-core):
  https://github.com/tueda/homebrew-loops/issues

These open issues may also help:
ci: brew test kira fails on Linux https://github.com/tueda/homebrew-loops/issues/20


I don't know if the linking will be correct since meson has been installed without arch --x86_64 or if this is not the case. 
If I try

brew86 link meson

I obtain the error:

Linking /usr/local/Cellar/meson/1.3.1... 
Error: Could not symlink share/vim/vimfiles/ftdetect
/usr/local/share/vim/vimfiles is not writable.

**Reply by SMillerDev** [reply]
`sudo chown -R $(whoami) /usr/local/share/vim/vimfiles` should fix the permission issue. The other error seems to be an issue with the compatibility between brew's meson version and Kira's install script though.

**Reply by MicheleD1999** [reply]
I tried with sudo chown -R $(whoami) /usr/local/share/vim/vimfiles and then brew86 link meson, and the tap seemed to work before stopping with a similar error:

brew86 install kira.rb                               
==> Downloading https://formulae.brew.sh/api/formula.jws.json
########################################################################################################################## 100.0%
==> Downloading https://formulae.brew.sh/api/cask.jws.json
########################################################################################################################## 100.0%
==> Fetching tueda/loops/kira
==> Downloading https://gitlab.com/kira-pyred/kira/-/archive/2.3/kira-2.3.tar.gz
Already downloaded: /Users/micheledradi/Library/Caches/Homebrew/downloads/912c4eee6ca2c949bdd867034cec44a3056762091f893b660f3bd475b558b2e4--kira-2.3.tar.gz
==> Installing kira from tueda/loops
==> meson --force-fallback-for=firefly ..
==> ninja
Last 15 lines from /Users/micheledradi/Library/Logs/Homebrew/kira/02.ninja:
      firefly::RatReconst::get_canonical_factors(std::__1::pair<std::__1::unordered_set<unsigned int, std::__1::hash<unsigned int>, std::__1::equal_to<unsigned int>, std::__1::allocator<unsigned int>>, std::__1::unordered_set<unsigned int, std::__1::hash<unsigned int>, std::__1::equal_to<unsigned int>, std::__1::allocator<unsigned int>>> const&) in RatReconst.cpp.o
  _nmod_poly_set_str, referenced from:
      firefly::RatReconst::get_canonical_factors(std::__1::pair<std::__1::unordered_set<unsigned int, std::__1::hash<unsigned int>, std::__1::equal_to<unsigned int>, std::__1::allocator<unsigned int>>, std::__1::unordered_set<unsigned int, std::__1::hash<unsigned int>, std::__1::equal_to<unsigned int>, std::__1::allocator<unsigned int>>> const&) in RatReconst.cpp.o
      firefly::RatReconst::get_canonical_factors(std::__1::pair<std::__1::unordered_set<unsigned int, std::__1::hash<unsigned int>, std::__1::equal_to<unsigned int>, std::__1::allocator<unsigned int>>, std::__1::unordered_set<unsigned int, std::__1::hash<unsigned int>, std::__1::equal_to<unsigned int>, std::__1::allocator<unsigned int>>> const&) in RatReconst.cpp.o
clang: error: linker command failed with exit code 1 (use -v to see invocation)
[27/65] Compiling C++ object src/pyred/libpyred.a.p/keyvaluedb.cpp.o
[28/65] Compiling C++ object subprojects/firefly/test/test_bunch.p/test_bunch.cpp.o
[29/65] Compiling C++ object subprojects/firefly/test/test_normal.p/test_normal.cpp.o
[30/65] Compiling C++ object subprojects/firefly/test/test_safe.p/test_safe.cpp.o
[31/65] Compiling C++ object subprojects/firefly/test/test_save_write.p/test_save_write.cpp.o
[32/65] Compiling C++ object subprojects/firefly/example.p/example.cpp.o
[33/65] Compiling C++ object src/pyred/libpyred.a.p/integrals.cpp.o
[34/65] Compiling C++ object src/pyred/libpyred.a.p/interface.cpp.o
[35/65] Compiling C++ object subprojects/firefly/ff_insert.p/ff_insert.cpp.o
ninja: build stopped: subcommand failed.

If reporting this issue please do so at (not Homebrew/brew or Homebrew/homebrew-core):
  https://github.com/tueda/homebrew-loops/issues

These open issues may also help:
ci: brew test kira fails on Linux https://github.com/tueda/homebrew-loops/issues/20


And if I try:
brew86 link ninja 
Linking /usr/local/Cellar/ninja/1.11.1... 
Error: Could not symlink share/emacs/site-lisp/ninja
/usr/local/share/emacs/site-lisp is not writable.
 
Should I continue with sudo chown -R $(whoami) and linking every brew installation to /usr/local?

**Comment by MicheleD1999** [comment]
Thank you very much for the support: following this path (linking with also --overwrite and permitting with sudo clown) I finally managed to install correctly with the x86_64 architecture.

Best wishes,
Michele
